include/lsst/afw/geom/detail/frameSetUtils.h

```diff
@@ -107,6 +107,20 @@ Copy values from an AST FitsChan into a PropertyList
 */
 std::shared_ptr<daf::base::PropertyList> getPropertyListFromFitsChan(ast::FitsChan& fitsChan);
 
+/**
+Construct AST FitsChan from PropertyList
+
+@param[in] metadata  Metadata to transfer; if this is a PropertyList then the order of items is preserved.
+@param[in] excludeNames  Names of entries to exclude from the returned header string.
+@param[in] options  Options string to pass to ast::FitsChan constructor.
+
+@return an ast::FitsChan representing this PropertyList
+
+*/
+std::shared_ptr<ast::FitsChan> getFitsChanFromPropertyList(daf::base::PropertySet& metadata,
+                                                           std::set<std::string> const& excludeNames = {},
+                                                           std::string options = "");
+
```

Comment:
I suggest you return a shared ptr instead of a FitsChan. Return value optimization should prevent copying and there is only one kind of FitsChan so polymorphism is not an issue. Or am I missing something?

Reply:
I was doing what was done with `getPropertyListFromFitsChan`.

Reply:
PropertyList is polymorphic so pointers are important. Not so for FitsChan.

Reply:
I withdraw the suggestion. Channels cannot be copied so it's safest not to pretend that's what we are doing here (even though it would not really be copied). I'm not even sure it would compile. What you have is likely better than what I suggested.

Reply:
I'm pretty sure this is a move, not a copy (and `FitsChan` does seem to be movable).  I think the argument that this would have been an elided copy would have been true in C++98 but is not true in C++11.

Reply:
@TallJimbo so do you have a preference? I think the pointer is a bit less obvious, but both work.

Reply:
Slight preference for returning by value.

Reply:
that would be dropping all the shared pointer stuff and returning `ast::FitsChan` directly?

Reply:
Yes.